tests/_auth_fakes.py

```
@@ -212,5 +214,94 @@ def save_jira_workspace_prefs(self, *, project_key: str) -> None:
+            return dict(instances.get(host, {}))
+        return {}
+
+    def save_gitlab_credentials(
+        self,
+        credentials: dict[str, Any],
+        *,
+        account: dict[str, Any] | None = None,
+    ) -> None:
+        pat = str(credentials.get("personal_access_token") or "").strip()
+        if not pat:
+            raise ProviderCredentialError("GitLab personal access token is required.")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)